### DISCUSSION TITLE
Add generic types for `useVueFlow`

### DISCUSSION THREAD

**Discussion by mdominiak2** [bcakmakoglu/vue-flow#1101] [Ideas]
I would like to type the Nodes. I haven't found a way to do this, so I propose generic types for `useVueFlow`.
Something like this `useVueFlow<NodaData, EdgeData, CustomEvents, KeyType>`.

**Comment by bcakmakoglu** [comment]
Mh... yeah I've thought about that too but then decided not to go that route atm.
You can still type nodes yourself though as they allow for generics, like `const nodes = ref<Node<NodeData, CustomEvents>[]>([...])`

**Reply by bcakmakoglu** [reply]
Might be I'll add this generic in 2.0.0 but I'm not sure yet 🤷‍♂️
